hashednode.go

```diff
@@ -31,14 +31,37 @@ import (
 )
 
 type HashedNode struct {
-	commitment  []byte
+	commitment  SerializedPointCompressed
```

Comment:
Here I just formalize the type to avoid confusion if `[]byte` is compressed or uncompressed.
`commitment` is the compressed serialization of the point. Same semantics as before, but being clear here.

We can't assume we have this value always available. The serialization of points now are in uncompressed form. Transforming that to compressed is CPU costly.

`commitment` will be calculated _if_ needed and in the fastest possible way depending on the context. 
More about this to be explained in next comments. The TL;DR: think of it as a "cached" value.